### PR TITLE
Add more tests to driver

### PR DESCRIPTION
**Pull request by pitag-ha**
The reason why the ocaml-ci fails on other ocaml versions than 4.10 is that at the moment the driver only accepts binary AST's of the same version the driver got compiled with; and in the test that's failing I'm handing in a 4.10 binary AST. Do you want me to remove that test for now to have a green CI and add it when that's fixed?

### CODE REVIEW

File: test/driver/error_embedding/dune [hunk 1]
```diff
@@ -0,0 +1,12 @@
+(executable
+ (public_name raiser)
+ (modules raiser)
+ (libraries ppxlib))
+
+(executable
+ (public_name pp)
+ (modules pp)
+ (libraries ppxlib))
```
Comment (jeremiedimino) [inline test/driver/error_embedding/dune:9]: Nitpick: the two above stanzas can be merged into a single `executables` stanza

File: test/driver/error_embedding/test.t/run.t [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  $ export OCAML_ERROR_STYLE=short
+
+With the `-embed-errors` options, if a PPX raises, the first such exception is caught and packed into an output AST
+
+  $ echo "let _ = [%raise]" > impl.ml
+  $ raiser -embed-errors impl.ml
+  [%%ocaml.error "Raising inside the rewriter"]
+
+The same is true when using the `-as-ppx- mode (note that the error is reported by ocaml itself)
```
Comment (jeremiedimino) [inline test/driver/error_embedding/test.t/run.t:11]:
```suggestion
The same is true when using the `-as-ppx` mode (note that the error is reported by ocaml itself)
```

File: test/driver/error_embedding/test.t/run.t [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  $ ocaml -ppx 'raiser -as-ppx' impl.ml
+  File "./impl.ml", line 1, characters 8-16:
+  Error: Raising inside the rewriter
+  [2]
+
+Also exceptions raised in a preprocessor get embedded into an AST (while the error from the preprocessor's stderr also gets reported on the driver's stderr)
```
Comment (jeremiedimino) [inline test/driver/error_embedding/test.t/run.t:18]: Could you make sure lines are no longer than 80 columns? Long lines make reading/reviewing the code more difficult. In the github UI, we have to scroll horizontally to read the full sentence.

**Comment by jeremiedimino** [comment]
> Do you want me to remove that test for now to have a green CI and add it when that's fixed?

Yes, please. It's easier when the CI is all green as we don't have to go through the tests to see which ones are failing.

**Review by jeremiedimino — CHANGES_REQUESTED** [review]
Good testing!

File: test/driver/error_embedding/dune (hunk 1 above)
Comment (pitag-ha) [reply test/driver/error_embedding/dune:3]: Nice! I didn't know that stanza.

File: test/driver/error_embedding/test.t/run.t (hunk 2 above)
Comment (pitag-ha) [reply test/driver/error_embedding/test.t/run.t]: For some reason github doesn't let me press commit suggestion. Anyways, done.

File: test/driver/error_embedding/test.t/run.t (hunk 3 above)
Comment (pitag-ha) [reply test/driver/error_embedding/test.t/run.t]: Ok!

**Review by jeremiedimino — APPROVED** [review]
Looks good!

**Pull request merged**